ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/encoders/TomlAbstractEncoder.kt

```diff
@@ -204,7 +213,15 @@ public abstract class TomlAbstractEncoder protected constructor(
 
     protected open fun isNextElementKey(descriptor: SerialDescriptor, index: Int): Boolean {
         when (val kind = descriptor.kind) {
-            StructureKind.CLASS -> setKey(descriptor.getElementName(index))
+            StructureKind.CLASS -> {
+                // We should keep previous key when we have value (inline) class
+                // But if key is null, it means that value class isn't nested, and we have to use its own key
+                if (descriptor.isInline && attributes.key != null) {
+                    // do nothing
```

Comment:
But why do we need this empty branch? 
May be have the following (if it is valid):
```
if (!descriptor.isInline || attributes.key == null)) {
    setKey(descriptor.getElementName(index))
}

```